### ISSUE TITLE
How to avoid NaN in training

### ISSUE THREAD

**Issue by frickyinn** [cleinc/bts#149]
https://github.com/cleinc/bts/blob/dd62221bc50ff3cbe4559a832c94776830247e2e/pytorch/bts.py#L47

Nan may occur while training because the final depth estimation sometimes yields 0 values. However, the loss function never tries to avoid log(0).
The solution to the situation is simply changing to this:
```python
d = torch.log(depth_est[mask] + 1e-4) - torch.log(depth_gt[mask]) 
```
